libfwupdplugin/fu-config.c

```diff
@@ -534,7 +534,11 @@ fu_config_reset_defaults(FuConfig *self, const gchar *section, GError **error)
 	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
 
 	/* remove all keys, and save */
-	g_key_file_remove_group(priv->keyfile, section, NULL);
+	if (!g_key_file_remove_group(priv->keyfile, section, error)) {
+		fwupd_error_convert(error);
+		return FALSE;
+	}
```

Comment:
not sure about this; if we call fu_config_reset_defaults() with the same section name -- does the second call fail?

Reply:
Yeah; if you call it twice it will fail the second time because the config is empty.

Reply:
Isn't that the opposite of what we want tho? i.e. if I reset something i want to make it a clean slate, not explicitly just undo the last action.

Reply:
It clears the whole section.  If the section is missing from the first clear call then it shouldn't be possible to clear again?

Reply:
Furthermore; if that is the behavior we want to allow (you can clear a missing section) then I think it's better to catch that specific error and waive it.

Reply:
Ack, agree.

Reply:
done